cloudmarker/util.py

```diff
@@ -118,18 +126,85 @@ def parse_cli(args=None):
         argparse.Namespace: Parsed command line arguments.
 
     """
-    parser = argparse.ArgumentParser(prog='cloudmarker')
-    parser.add_argument('-c', '--config', nargs='+',
-                        default=['config.base.yaml', 'config.yaml'],
+    default_config_paths = [
+        '/etc/cloudmarker.yaml',
+        '~/.cloudmarker.yaml',
+        '~/cloudmarker.yaml',
+        'cloudmarker.yaml',
+    ]
+
+    description = """
+    Audit clouds as specified by configuration.
+
+    Zero or more config files are specified with the -c/--config option.
+    The config files specified are merged with a built-in base config.
+    Use the -p/--print-base-config option see the built-in base config.
+    Missing config files are ignored.
+
+    If two or more config files provide conflicting config values, the
+    config file specified later overrides the built-in base config and
+    the config files specified earlier.
+
+    If the -c/--config option is specified without any file arguments
+    following it, then only the built-in base config is used.
+
+    If the -c/--config option is omitted, then the following config
```

Comment:
If -c/--config option is specified with file arguments then the code tries to read those files and load the configuration from them. In case the file is non-existent the file is ignored. Shouldn't it gracefully exit in case a user specified file is missing?  

Whereas, if -c/--config option is omitted it should look for config files in popular location, load and merge them with the built-in base config.

Reply:
This sounds like a good idea. This will add a little bit of complexity to the `parse_cli()` and `load_config()` functions, so I will create a separate pull request for it, so that we can discuss it separately and see if this feature is important enough to introduce this little bit of complexity.